Lib/glyphsLib/featureWriters/markFeatureWriter.py

```diff
@@ -230,18 +230,27 @@ def _makeFeatures(self):
                 lookup.statements.append(MarkToBasePos(glyph, [anchor]).asAST())
             lookups.append(lookup)
 
+            # Insert mark glyph names after base glyph names if not specified otherwise.
+            if "&" not in after:
+                after = after.replace("*", "* &")
+
+            # Group base glyphs by anchor
+            glyphs = {}
             for glyph, anchor in glyph_anchor_pair:
+                glyphs.setdefault(anchor.key, [anchor, []])[1].append(glyph)
+
+            for anchor, bases in glyphs.values():
+                bases = " ".join(bases)
                 marks = ast.GlyphClass(
                     self.context.markClasses[anchor.key].glyphs.keys()
                 ).asFea()
-                if "&" not in after:
-                    after = after.replace("*", "* &")
-                # Replace & with mark name if present
-                contextual = after.replace("*", f"{glyph}")
+
+                # Replace * with base glyph names
+                contextual = after.replace("*", f"[{bases}]")
```

Comment:
should we avoid the enclosing `[]` if `bases` happens to contain only one glyph maybe? or it doesn't actually make any difference in the built output?

Reply:
It does not seem to make a difference, we are getting the same lookup type (8, chained contexts positioning) with and without this PR.